test cases/common/179 source in dep/generated/meson.build

```diff
@@ -0,0 +1,12 @@
+fp = find_program('genheader.py')
+
+genh = custom_target('genh',
+  input : 'funname',
+  output : 'funheader.h',
+  command : [fp, '@INPUT@', '@OUTPUT@'])
+  
+dep = declare_dependency(sources : [genh])
+  
+e = executable('genuser', 'main.c',
+  dependencies : dep)
+test('genuser', e)
```

Comment:
Shouldn't these two be in separate directories so that we don't implicitly include the current builddir and make it pass?

Reply:
Builddirness is tested elsewhere, the test in fact requires that the file is generated in the current build dir . This test is about making sure the header file is generated before compiling the source file.